File: iossdkdemo/SDKDemo/APIControllers/UploadMediaViewController.m
//
//  UploadMediaViewController.m
//  SDKDemo
//
//  Created by OnePoint Global on 04/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

#import "UploadMediaViewController.h"
#import <OPGSDK/OPGSDK.h>

@interface UploadMediaViewController ()

@property(strong, nonatomic) NSString* mediaID;

@end

@implementation UploadMediaViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)imagePickerController:(UIImagePickerController *)picker
        didFinishPickingImage:(UIImage *)image
                  editingInfo:(NSDictionary *)editingInfo
{
    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory,     NSUserDomainMask, YES);
    NSString *documentsDirectory = [paths objectAtIndex:0];
    NSString *savedImagePath = [documentsDirectory stringByAppendingPathComponent:@"savedImage.png"];
    
    OPGSDK* sdk = [OPGSDK new];                   // Creating OPGSDK instance
    NSError *error;
    self.mediaID = [sdk uploadMediaFile:savedImagePath error:&error];
    NSLog(@"Uploaded media ID is %@",_mediaID);
    
    [self dismissViewControllerAnimated:YES completion:nil];
    
}

#pragma mark - IBAction
-(IBAction)uploadImage:(id)sender
{
    UIImagePickerController *imagePickerController = [[UIImagePickerController alloc]init];
    imagePickerController.delegate = self;
    imagePickerController.sourceType =  UIImagePickerControllerSourceTypePhotoLibrary;
    [self presentViewController:imagePickerController animated:YES completion:nil];
}


@end

File: iossdkdemo/SDKDemo-Swift/APIControllers-Swift/UploadMediaViewController.swift
//
//  UploadMediaViewController.swift
//  SDKDemoApp
//
//  Created by OnePoint Global on 04/10/16.
//  Copyright © 2016 opg. All rights reserved.
//

import UIKit

class UploadMediaViewController: UIViewController,UIImagePickerControllerDelegate, UINavigationControllerDelegate
{

    var mediaID : NSString?
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    
    @IBAction func uploadMediaSwift(_ sender: AnyObject)
    {
        let imagePicker = UIImagePickerController()
        imagePicker.allowsEditing = false
        imagePicker.delegate=self
        imagePicker.sourceType = .photoLibrary
        
        present(imagePicker, animated: true, completion: nil)
    }
    
    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingImage image: UIImage, editingInfo: [String : AnyObject]?)
    {
        let writePath = URL(fileURLWithPath: NSTemporaryDirectory()).appendingPathComponent("savedImage.png")
        
        let sdk = OPGSDK()        // Creating OPGSDK instance
        
        do {
            mediaID = try sdk.uploadMediaFile(writePath.absoluteString) as NSString?
        }
        catch{
            print("Upload Media Media Failed")         /* @"Error Occured. Contact Support!" */
            
        }
        print("Uploaded media ID is \(mediaID)" )
        self.dismiss(animated: true, completion: nil);
    }

}
